diff --git a/hardware/nvidia/t23x/nv-public/overlay/Makefile b/hardware/nvidia/t23x/nv-public/overlay/Makefile
index 93728f4..cce4d96 100644
--- a/hardware/nvidia/t23x/nv-public/overlay/Makefile
+++ b/hardware/nvidia/t23x/nv-public/overlay/Makefile
@@ -13,6 +13,7 @@ dtbo-y += tegra-optee.dtbo
 dtbo-y += tegra234-audio-overlay.dtbo
 dtbo-y += tegra234-carveouts.dtbo
 dtbo-y += tegra234-dcb-p3767-0000-hdmi.dtbo
+dtbo-y += tegra234-kepler-ecg-v1-dynamic.dtbo
 dtbo-y += tegra234-p3737-0000+p3701-0000-as-p3701-0004.dtbo
 dtbo-y += tegra234-p3737-0000+p3701-0000-as-p3767-0000.dtbo
 dtbo-y += tegra234-p3737-0000+p3701-0000-as-p3767-0001.dtbo
